### DISCUSSION TITLE
Return normal vector given a workplane?

### DISCUSSION THREAD

**Discussion by xpe** [CadQuery/cadquery#1522] [General]
Given a `Workplane` how can one get its normal vector? This feels like a very natural thing to do, but I don't see how. 

Is my question sensical given the definitions of CadQuery? Why or why not?

In `cadquery.occ_impl.shapes` I see `normal` and `normalAt` but they don't apply to `Workplane`:

```python
class Mixin1D(object):

# ...

def normal(self: Mixin1DProtocol) -> Vector:
        """
        Calculate the normal Vector. Only possible for planar curves.

        :return: normal vector
        """
```

```python
class Face(Shape):
    """
    a bounded surface that represents part of the boundary of a solid
    """
    
# ...

def normalAt(self, locationVector: Optional[Vector] = None) -> Vector:
        """
        Computes the normal vector at the desired location on the face.

        :returns: a vector representing the direction
        :param locationVector: the location to compute the normal at. If none, the center of the face is used.
        :type locationVector: a vector that lies on the surface.
        """
```

**Comment by lorenzncode** [comment]
You can inspect the Workplane `Plane` object.

```py
mywp = cq.Workplane("XZ")
print(mywp.plane)
```

**Reply by xpe** [reply]
Ok, thanks. I see `print(mywp.plane)` shows e.g.

```
Plane(origin=(0.0, 0.0, 8.900000000000002), xDir=(1.0, 0.0, 0.0), normal=(0.0, 0.0, 1.0))
```

However, can the normal vector be accessed programmatically?

1. I tried: `print(mywp.plane.normal)`

2. I see no mention of a `normal` property [in the source code for the `Plane` class][1].

What's the story... what am I missing?

Ah:

```python
def __repr__(self):
    return f"Plane(origin={str(self.origin.toTuple())}, xDir={str(self.xDir.toTuple())}, normal={str(self.zDir.toTuple())})"
```

So the answer is the `zDir` property on `Plane`.


[1]: https://github.com/CadQuery/cadquery/blob/551a23157026faef423e7cb7caaf6f84da91a562/cadquery/occ_impl/geom.py#L393-L777

**Reply by lorenzncode** [reply]
>So the answer is the zDir property on Plane.

right

```py
normal = mywp.plane.zDir.toTuple()
```

You can also define your own plane.
```py
plane0 = cq.Plane((0, 0, 0), normal=(1, 1, 1))
res = cq.Workplane(plane0).circle(1).extrude(4)
```
